MeretzLib/Apple/iOS/MeretzSampleGame/MeretzSampleGame/GameViewController.m

```diff
@@ -8,351 +8,766 @@
-				if (MeretzTaskStatusComplete == taskStatus)
+				if (MERETZ_TASK_ID_INVALID != self.VendorDisconnectUserTaskId)
 				{
-					MeretzVendorConsumeResult *consumeResults= [gMeretz getVendorConsumeResult:gVendorConsumeTaskId];
-					
-					NSLog(@"VendorConsume results: %@", consumeResults);
-					
-					// release our /vendor/consume task results
-					[gMeretz releaseTask:gVendorConsumeTaskId];
-					gVendorConsumeTaskId= MERETZ_TASK_ID_INVALID;
+					MeretzTaskStatus taskStatus= [MeretzAPI getTaskStatus:self.VendorDisconnectUserTaskId];
 					
-					// initiate a /vendor/disconnect
-					gVendorDisconnectTaskId= [gMeretz vendorUserDisconnect];
-					
-					if (MERETZ_TASK_ID_INVALID != gVendorDisconnectTaskId)
+					if (MeretzTaskStatusComplete == taskStatus)
 					{
-						gSampleGameState= Disconnecting;
+						MeretzResult *disconnectResults= [MeretzAPI getVendorUserDisconnectResult:self.VendorDisconnectUserTaskId];
+						
+						[self didVendorUserDisconnectFinish:disconnectResults];
 					}
 					else
 					{
-						NSLog(@"failed to initiate a /vendor/disconnect call!");
-						gSampleGameState= Finished;
+						NSLog(@"VendorDisconnect task still in progress...");
 					}
 				}
-				else
+				else if (!USE_MERETZ_DELEGATE)
 				{
-					NSLog(@"VendorConsume task still in progress...");
+					NSLog(@"Lost our VendorUserDisconnect task?");
+					self.SampleGameState= Idle;
 				}
+				break;
+			}
+			case Idle:
+			default:
+			{
+				break;
+			}
+		}
+		
+		return;
+	}
+
+	-(NSString *) getSavedMeretzAccessToken
+	{
+		NSString *savedAccessToken= [[NSUserDefaults standardUserDefaults] stringForKey:MERETZ_USER_ACCESS_TOKEN_NSUSERDEFAULTS_KEY];
+	
+		return savedAccessToken;
+	}
+
+	-(void) saveMeretzAccessToken: (NSString *)accessToken
+	{
+		[[NSUserDefaults standardUserDefaults] setObject:accessToken forKey:MERETZ_USER_ACCESS_TOKEN_NSUSERDEFAULTS_KEY];
+		[[NSUserDefaults standardUserDefaults] synchronize];
+		
+		return;
+	}
+
+	-(void) beginVendorUserConnect
+	{
+		// initiate a user connection task
+		NSString *userConnectionCode= self.ConnectionCodeTextField.text;
+		NSString *vendorUserIdentifier= SAMPLE_GAME_USER_IDENTIFIER;
+		
+		if (USE_MERETZ_DELEGATE)
+		{
+			if (![MeretzAPI startVendorUserConnect:userConnectionCode vendorUserToken:vendorUserIdentifier])
+			{
+				NSLog(@"startVendorUserConnect failed!");
+			}
+		}
+		else
+		{
+			MeretzTaskId result= [MeretzAPI vendorUserConnect:userConnectionCode vendorUserToken:vendorUserIdentifier];
+			
+			if (MERETZ_TASK_ID_INVALID != result)
+			{
+				NSLog(@"initiating vendorUserConnect with connection code '%@'", userConnectionCode);
+				self.SampleGameState= Connecting;
 			}
 			else
 			{
-				gSampleGameState= Finished;
+				NSLog(@"failed to initiate vendorUserConnect!");
+				self.SampleGameState= Idle;
 			}
-			break;
+			
+			self.ConnectionCodeTextField.text= @"";
+			
+			self.VendorConnectUserTaskId= result;
 		}
-		case Disconnecting:
+		
+		return;
+	}
+
```

Comment:
logic to consume a user's newly acquired items from Meretz (either high-level API or low-level API)